hw/dv/dpi/usbdpi/usbdpi.c

```diff
@@ -383,45 +395,37 @@ void readDescriptor(struct usbdpi_ctx *ctx) {
 }
 
 // Get Baud
-void readBaud(struct usbdpi_ctx *ctx) {
+void readBaud(usbdpi_ctx_t *ctx) {
+  usbdpi_transfer_t *tr = ctx->sending;
+  uint8_t *dp;
   switch (ctx->hostSt) {
     case HS_STARTFRAME:
-      ctx->state = ST_SYNC;
-      ctx->bytes = 14;
-      ctx->datastart = 3;
-      ctx->byte = 0;
-      ctx->bit = 1;
-      ctx->data[0] = USB_PID_SETUP;
-      ctx->data[1] = 0x82;
-      ctx->data[2] = 0 | CRC5(0x82, 11) << 3;
-      ctx->data[3] = USB_PID_DATA0;
-      ctx->data[4] = 0xc2;  // d2h, vendor, endpoint
-      ctx->data[5] = 2;     // get baud
-      ctx->data[6] = 0;     // index 0
-      ctx->data[7] = 0;     // type device
-      ctx->data[8] = 0;     // wIndex = 0
-      ctx->data[9] = 0;
-      ctx->data[10] = 0x2;  // wLength = 2
-      ctx->data[11] = 0;
-      add_crc16(ctx->data, ctx->datastart, 12);
-      // ctx->data[12] = 0x10; // pre-computed CRC32
-      // ctx->data[13] = 0xDD;
+      transfer_token(tr, USB_PID_SETUP, ctx->dev_address, ENDPOINT_ZERO);
+
+      dp = transfer_data_start(tr, USB_PID_DATA0, 0);
+      dp[0] = 0xc2;  // d2h, vendor, endpoint
+      dp[1] = 2;     // get baud
+      dp[2] = 0;     // index 0
+      dp[3] = 0;     // type device
+      dp[4] = 0;     // wIndex = 0
+      dp[5] = 0;
+      dp[6] = 0x2;  // wLength = 2
+      dp[7] = 0;
+      transfer_data_end(tr, dp + 8);
+
+      // TODO - why are we not setting bus_state here too?
```

Comment:
Because this existed before I added the "bus state," to cobble together some control transfers without modifying too much of the code, haha. I stopped adding modifications after fixing up the standard device configuration parts, and I ignored the serial port-specific ones. Oops!

...plus, this code was broken anyway.

Reply:
Thanks for the clarifications and guidance. Re the exported diags...in the next batch of work the state variables and PIDs have been changed to enums which means that waveforms in GTKWave (and I guess other environments) show symbolic names rather than numeric values.